catalog/imaging/imaging-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPostIngestPlugin.java

```diff
@@ -127,6 +131,8 @@
 
   private NitfParserService nitfParserService;
 
+  private Semaphore lock = new Semaphore(MAX_THREAD_COUNT, true);
```

Comment:
Not sure what unit tests I'd write. It is generally not worth it to write unit tests that test threaded functionality. Can you elaborate what you'd want to see for unit tests?

Reply:
I was thinking a test that verifies that only so many threads could be processing at any given time. I wasn't thinking that would be a very complex unit test.

Reply:
Well it'd be testing that the lock is working as it is intended, which is now a major component of this class's behavior. Someone could refactor this class and unintentionally change the boundary of the lock so that the processing happens outside the locking mechanism. I know that's unlikely, but that's the sort of thing the unit tests are meant to guard against. If you think it'd be too much effort, I won't fight it 😄 I _do_ think it'd be good to get another opinion.

Reply:
Well that is why I'm talking about it on here instead of just bugging you directly :)

Reply:
It isn't something I think would be that easy to test, to make sure that you've got the right stuff inside the lock and that the lock is restricting the number of threads like we expect. I am certainly open to ideas, if folks have a way to easily test out that sort of thing in a unit test.